### PR TITLE
Update engine testing module and events test

### PR DESCRIPTION
**Pull request by roronoasins**
|Related issue|
|-------------|
|      https://github.com/wazuh/wazuh-qa/issues/3474      |

## Description

This PR aims to update the engine testing module and the events test that verify that the engine generates the expected alerts.

### Added

- `integration/test_engine/conftest.py`

### Updated

- `modules/engine/__init__.py`
- `modules/engine/event_monitor.py`
- `tools/__init__.py`
- `test_events/data/test_cases/cases_engine_events.yaml`
- `test_events/test_events.py`

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @roronoasins (Developer)  |    `integration/test_engine/test_events/test_events.py`       | :no_entry_sign: :no_entry_sign::no_entry_sign:  |     :green_circle: :green_circle: :green_circle: :green_circle: :green_circle:    |    Ubuntu 20.04     | Nothing to highlight | Blocked by https://github.com/wazuh/wazuh-jenkins/issues/4674

### CODE REVIEW

File: tests/integration/test_engine/test_events/test_events.py [hunk 1]
```diff
@@ -1,83 +1,97 @@
...
+    - wazuh-engine
 
-    yield
+os_platform:
+    - linux
 
-    print('INFO: Removing KVDB files')
-    shutil.rmtree('/tmp/win-security-categorization/', ignore_errors=False, onerror=None)
+os_version:
+    - Ubuntu Focal
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:27]: What about Centos, ALAS, Debian, Suse, RHEL? Is it only for Ubuntu Focal?

File: tests/integration/test_engine/test_events/test_events.py [hunk 2]
```diff
@@ -1,83 +1,97 @@
...
-    print(f"INFO: Sending encoded event: {encoded_msg}")
 
 @pytest.mark.tier(level=0)
-@pytest.mark.parametrize('event, expected_output', zip(events, expected_outputs))
-def test_engine_events(create_kvdb, start_engine, send_events_to_the_engine, expected_output):
-    print(f"expected_output: {type(expected_output)}")
-    print(f"waiting for {expected_output[2:len(expected_output)-1]}")
-    evm.check_engine_json_event(event=expected_output[2:len(expected_output)-1])
+@pytest.mark.parametrize('events_data', t1_configuration_metadata, ids=t1_case_ids)
+def test_engine_events(events_data, truncate_engine_files, restart_engine_function):
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:58]: - Doesn't it require any extra configuration?
- Could you change the name to another descriptive one, please? The word events is too general. This case is about a specific case of input.

File: tests/integration/test_engine/test_events/test_events.py [hunk 3]
```diff
@@ -1,83 +1,97 @@
...
+    '''
+    description: Check that every event sent is correctly
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:60]: Refine the description to see the difference with the new cases to create.

File: tests/integration/test_engine/test_events/test_events.py [hunk 4]
```diff
@@ -1,83 +1,97 @@
...
+
+    test_phases:
+        - Clean the log and alert files
+        - Restart the wazuh-engine.
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:64]:
```suggestion
        - Restart the wazuh-engine
```

File: tests/integration/test_engine/test_events/test_events.py [hunk 5]
```diff
@@ -1,83 +1,97 @@
...
+            brief: Restart the wazuh-engine daemon.
+
+    assertions:
+        - Verify that after sending events we are allowed to catch them within the engine alerts.
+
+    input_description:
+        - The `cases_engine_events.yaml` file provides the module configuration for this test.
+
+    expected_output:
+        - Every item within the metadata.engine_outputs object
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:90]: Improve the expected_output.

File: tests/integration/test_engine/test_events/test_events.py [hunk 6]
```diff
@@ -1,83 +1,97 @@
...
+    '''
+    # Send the messages
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:92]: Improve the comment to add relevant information.

**Review by damarisg — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 7]
```diff
@@ -1,79 +1,40 @@
...
+    # Create a unixgram socket instance
+    events_socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
 
-    # msg format -> queue:location_str:msg
-    msg_formatted = QUEUE + ':' + LOCATION + ':' + event
-    msg_tam = len(msg_formatted)
-    msg_tam_little_endian = struct.pack('<I', msg_tam)
+    for event in events:
+        # Build the message with the expected format: {queue:location_str:msg}
+        msg_formatted = (QUEUE + ':' + LOCATION + ':' + event).encode('utf8')
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py:37]: `QUEUE` ad `LOCATION` store a default string for testing purposes to be sent with the events, so the user can modify them as they want. They will be added as parameters with that value as default.

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 8]
```diff
@@ -137,8 +137,8 @@ def get_service():
 CLUSTER_SOCKET_PATH = os.path.join(WAZUH_PATH, 'queue', 'cluster')
 
 
-ANALYSISD_ANALISIS_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'analysis')
-ANALYSISD_QUEUE_SOCKET_PATH = os.path.join(QUEUE_SOCKETS_PATH, 'queue')
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/tools/__init__.py:141]: As analysisd will be replaced in 5.0 by the engine, these are no longer required and they were replaced by the engine's

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 9]
```diff
@@ -159,9 +159,9 @@ def get_service():
     'wazuh-apid': [],
     'wazuh-agentlessd': [],
     'wazuh-csyslogd': [],
-    'wazuh-analysisd': [
-                        ANALYSISD_ANALISIS_SOCKET_PATH,
-                        ANALYSISD_QUEUE_SOCKET_PATH
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/tools/__init__.py:164]: As analysisd will be replaced in 5.0 by the engine, these are no longer required and they were replaced by the engine's

File: deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py [hunk 10]
```diff
@@ -17,17 +21,28 @@ def make_engine_callback(pattern, prefix=engine.ENGINE_PREFIX):
         lambda: function that returns if there's a match in the file
 
     Examples:
-        >>> callback_bionic_update_started = make_vuln_callback("Starting Ubuntu Bionic database update")
+        >>> callback_api_connection = make_vuln_callback("Engine API endpoint: ... [{API_SOCKET_PATH}]")
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py [hunk 11]
```diff
@@ -17,17 +21,28 @@ def make_engine_callback(pattern, prefix=engine.ENGINE_PREFIX):
...
+        event (str): event to check within the engine alerts
+        error_message (str): error message to show in case of expected event does not occur
+        update_position (boolean): filter configuration parameter to search in Wazuh log
+        timeout (str): timeout to check the engine alerts
+        prefix (str): event pattern regex
+        accum_results (int): Accumulation of matches.
+    """
     file_monitor = FileMonitor(file_to_monitor) if file_monitor is None else file_monitor
     error_message = f"Could not find this event in {file_to_monitor}: {event}" if error_message is None else \
         error_message
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/conftest.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    """Truncate all the log files and json alerts files before and after the test execution."""
+    log_files = [engine.ENGINE_ALERTS_PATH, engine.ENGINE_LOG_PATH]
+
+    for log_file in log_files:
+        truncate_file(log_file)
+
+    yield
+
+    for log_file in log_files:
+        truncate_file(log_file)
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/conftest.py]: We will wait for the pending PRs to be merged before we can use the new accessories.

File: tests/integration/test_engine/conftest.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import pytest
+
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.modules import engine
+
+
+@pytest.fixture(scope='function')
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/conftest.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/conftest.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+def restart_engine_function():
+    """Start the wazuh-engine daemon before running a test, and stop it when finished."""
+    control_service('restart', daemon=engine.MODULE_NAME)
+
+    yield
+
+    control_service('stop', daemon=engine.MODULE_NAME)
+
+
+@pytest.fixture(scope='function')
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/conftest.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/test_events/test_events.py (hunk 1 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py:27]: As the engine has only been tested on this operating system, this list will be updated as we test it on more systems.

File: tests/integration/test_engine/test_events/test_events.py (hunk 2 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

About the configuration, it is not. The engine does not need any other daemon or process to run. We handle it with the restart fixture.

File: tests/integration/test_engine/test_events/test_events.py (hunk 4 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/test_events/test_events.py (hunk 3 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/test_events/test_events.py (hunk 5 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/commit/351e82e0a94e43f5bd1ae2cd2cb5e243f24cab27

File: tests/integration/test_engine/test_events/test_events.py (hunk 6 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/3e8f80721cd2affafd2a542fc917e898f083a4d1

File: tests/integration/test_engine/test_events/test_events.py [hunk 15]
```diff
@@ -1,83 +1,97 @@
...
 @pytest.mark.tier(level=0)
-@pytest.mark.parametrize('event, expected_output', zip(events, expected_outputs))
-def test_engine_events(create_kvdb, start_engine, send_events_to_the_engine, expected_output):
-    print(f"expected_output: {type(expected_output)}")
-    print(f"waiting for {expected_output[2:len(expected_output)-1]}")
-    evm.check_engine_json_event(event=expected_output[2:len(expected_output)-1])
+@pytest.mark.parametrize('events_data', t1_configuration_metadata, ids=t1_case_ids)
+def test_receiving_events_socket(events_data, truncate_engine_files, restart_engine_function):
+    '''
+    description: Check that every event sent through the engine's queue socket is correctly received and stored in the expected log files.
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:60]: Check indentation.

File: tests/integration/test_engine/test_events/test_events.py [hunk 16]
```diff
@@ -1,83 +1,97 @@
...
+            brief: Restart the wazuh-engine daemon.
+
+    assertions:
+        - Verify that after sending events we are allowed to catch them within the engine alerts.
+
+    input_description:
+        - The `cases_engine_events.yaml` file provides the module configuration for this test.
+
+    expected_output:
+        - Every item within the metadata.engine_outputs object should be placed within the expected log files
```
Comment (damarisg) [inline tests/integration/test_engine/test_events/test_events.py:90]:
```suggestion
        - Every item within the metadata.engine_outputs object should be placed within the expected log files.
```

**Review by damarisg — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 17]
```diff
@@ -1,79 +1,42 @@
-import sys
-import time
+# Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 18]
```diff
@@ -1,79 +1,42 @@
...
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
 import socket
-import struct
-import subprocess
+
+from wazuh_testing.tools import ENGINE_QUEUE_SOCKET_PATH
 
 
 # engine timeouts
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 19]
```diff
@@ -1,79 +1,42 @@
...
-T_1 = 1
+T_1 = 0.5
 
 # engine vars
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 20]
```diff
@@ -1,79 +1,42 @@
...
-ENGINE_OUTPUT_PATH = '/tmp/filepath.txt'
+ENGINE_ALERTS_PATH = '/var/ossec/logs/alerts/alerts-ECS.json'  # the engine uses this file during the dev phase
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 21]
```diff
@@ -1,79 +1,42 @@
...
-        # pid = subprocess.Popen([sys.executable, "longtask.py"], creationflags=DETACHED_PROCESS).pid
-    else:
-        subprocess_engine = subprocess.Popen(['/bin/bash', '-c', command], stdout=subprocess.PIPE, close_fds=True)
 
-    # wait till the engine is up
-    time.sleep(1)
+def send_events_to_engine_dgram(queue=QUEUE, location=LOCATION, events=[]):
+    """Send events to the engine events' socket.
 
+    The messages must follow the following format: queue:location_str:msg
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 22]
```diff
@@ -1,79 +1,42 @@
...
 
-def kill_engine():
-    """Kill the engine subprocess."""
-    print(f"INFO: Terminating the engine subprocess {subprocess_engine}")
-    subprocess_engine.kill()
+    The socket's protocol is unixgram, so we just need to send the events after formatting and encoding them.
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py:28]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 23]
```diff
@@ -1,79 +1,42 @@
...
+    """
+    # Create a unixgram socket instance
+    events_socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
 
-    # msg format -> queue:location_str:msg
-    msg_formatted = QUEUE + ':' + LOCATION + ':' + event
-    msg_tam = len(msg_formatted)
-    msg_tam_little_endian = struct.pack('<I', msg_tam)
+    for event in events:
+        # Build the message with the expected format: {queue:location_str:msg}
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 24]
```diff
@@ -1,79 +1,42 @@
...
+        msg_formatted = (queue + ':' + location + ':' + event).encode('utf8')
 
-    # the engine's tcpEndpoint expects: header(msg size(little endian)) + event with the expected format
-    encoded_msg = msg_tam_little_endian + msg_formatted.encode()
-    client_socket.send(encoded_msg)
-    print(f"INFO: Sending encoded event: {encoded_msg}")
+        # Send the encoded message to the engine's events socket
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py [hunk 25]
```diff
@@ -1,79 +1,42 @@
...
-        raise Exception(f"The command {command} returned {result_code} as result code.")
-
-def send_event_to_engine(event):
-    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-
-    client_socket.connect(('127.0.0.1', 5054))
+    Args:
+        queue(str): queue string that is used to create the message that will be sent to the socket
+        location(str): location string that is used to create the message that will be sent to the socket
+        events (list): Events that will be sent to the socket
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py [hunk 26]
```diff
@@ -1,3 +1,7 @@
+# Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/__init__.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: tests/integration/test_engine/conftest.py [hunk 27]
```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/conftest.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: tests/integration/test_engine/test_events/test_events.py [hunk 28]
```diff
@@ -1,83 +1,97 @@
...
-import pytest
-import socket
-import struct
-import shutil
-
-from wazuh_testing.tools.configuration import get_test_cases_data
-from wazuh_testing.modules.engine import event_monitor as evm
-from wazuh_testing.modules import engine
+'''
+copyright: Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: tests/integration/test_engine/test_events/test_events.py (hunk 16 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

File: tests/integration/test_engine/test_events/test_events.py (hunk 15 above)
Comment (roronoasins) [reply tests/integration/test_engine/test_events/test_events.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/0790776dcbd3179dc1af3c815178396a8e95f944

**Review by damarisg — CHANGES_REQUESTED** [review]
Currently, we have this schema:

```
 test_engine
   |- test_events
        |- test_events
             |- test_engine_events.py
```

The "Events" name is too generic. 
In the future, we could add more tests about it. For that reason, the categories should be easy to identify. 
I show an easy example. Please rename it according to your criteria, and try to improve it.

```
test_engine
  |- test_events
       |- test_input
            |- test_type_events.py
```

File: CHANGELOG.md [hunk 29]
```diff
@@ -2,6 +2,16 @@
...
 All notable changes to this project will be documented in this file.
 
+## [5.0.0] - Development (unreleased)
+
+Wazuh commit: TBD \
+Release report: TBD
+
+### Added
+
+- Add engine event tests. ([#3698](https://github.com/wazuh/wazuh-qa/pull/3698)) \- (Framework + Tests)
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply CHANGELOG.md]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

File: deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py [hunk 30]
```diff
@@ -17,20 +21,32 @@ def make_engine_callback(pattern, prefix=engine.ENGINE_PREFIX):
...
 
     return lambda line: regex.match(line) is not None
 
-def check_engine_json_event(file_monitor=None, event='', error_message=None, update_position=True,
+
+def check_engine_event_output(file_monitor=None, event='', error_message=None, update_position=True,
                               timeout=engine.T_1, accum_results=1, prefix=engine.ENGINE_PREFIX,
-                              file_to_monitor=engine.ENGINE_OUTPUT_PATH):
+                              file_to_monitor=engine.ENGINE_ALERTS_PATH):
+    """Check if a vulnerability event occurs
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/modules/engine/event_monitor.py]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

File: tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml [hunk 31]
```diff
@@ -0,0 +1,94 @@
+- name: Verify AWS logs
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

File: tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+    events:
+      - "\u0054\u0045\u0053\u0054\u0049\u004E\u0047"
+      - "a log \0with unicode null"
+      - "a log \nwith new line"
+    engine_outputs:
+      - '{"wazuh":{"queue":49,"origin":"location"},"event":{"original":"TESTING"}}'
+      - '{"wazuh":{"queue":49,"origin":"location"},"event":{"original":"a log "}}'
+      - '{"wazuh":{"queue":49,"origin":"location"},"event":{"original":"a log \\nwith new line"}}'
+
+- name: cisco ios logs
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

File: tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+  description: Verify that the engine accepts elasticsearch sample logs
+  configuration_parameters:
+    cfg: None
+  metadata:
+    events:
+      - '[2018-05-17T08:29:12,177][INFO ][o.e.c.m.MetaDataCreateIndexService] [vWNJsZ3] [test-filebeat-modules] creating index, cause [auto(bulk api)], templates [test-filebeat-modules], shards [5]/[1], mappings [doc]'
+    engine_outputs:
+      - '{"wazuh":{"queue":49,"origin":"location"},"event":{"original":"\[2018-05-17T08:29:12,177]\[INFO ]\[o.e.c.m.MetaDataCreateIndexService] \[vWNJsZ3] \[test-filebeat-modules] creating index, cause \[auto\(bulk api\)], templates \[test-filebeat-modules], shards \[5]/\[1], mappings \[doc]"}}'
+
+- name: auditd logs
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

File: tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml [hunk 34]
```diff
@@ -0,0 +1,94 @@
...
+      - 2019-11-27 15:09:34,721 | INFO  | Connector stomp stopped | org.apache.activemq.broker.TransportConnector | main
+      - 2019-11-27 15:09:34,722 | INFO  | Connector mqtt stopped | org.apache.activemq.broker.TransportConnector | main
+      - 2019-11-27 15:09:34,723 | INFO  | Connector ws stopped | org.apache.activemq.broker.TransportConnector | main
+      - 2019-11-27 15:09:34,727 | INFO  | PListStore:[/opt/activemq/data/localhost/tmp_storage] stopped | org.apache.activemq.store.wawa.plist.PListStoreImpl | main
+      - 2019-11-27 15:09:34,728 | INFO  | Stopping async queue tasks | org.apache.activemq.store.wawa.KahaDBStore | main
+      - 2019-11-27 15:09:34,730 | INFO  | Stopping async topic tasks | org.apache.activemq.store.wawa.KahaDBStore | main
+      - 2019-11-29 10:59:49,515 | INFO  | No Spring WebApplicationInitializer types detected on classpath | /admin | main
+      - INFO  | anonymous called org.apache.activemq.broker.jmx.QueueView.retryMessages[] at 27-11-2019 08:45:57,213 | qtp443290224-47
+      - INFO  | admin called org.apache.activemq.broker.jmx.QueueView.retryMessages[] at 27-11-2019 08:45:57,229 | qtp443290224-45
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_engine/test_events/data/test_cases/cases_engine_events.yaml]: Solved https://github.com/wazuh/wazuh-qa/pull/3698/commits/732043441ad7661cb0f22f379185ad50281031d7

**Review by damarisg — APPROVED** [review]
I approved it. However, it will remain blocked until https://github.com/wazuh/wazuh-jenkins/issues/4674 finishes.

**Pull request closed without merging**